src/java/com/fulcrumgenomics/util/DemuxFastqs.java

```diff
@@ -687,20 +681,53 @@ private String getMolecularBarcode(final List<FastqRecord>  records) {
         }
     }
 
-    /** Reads multiple FASTQs at the same time, ensuring their read names match along the way */
-    class MultiFastqReader implements Iterator<List<FastqRecord>> {
-        public final List<FastqReader> readers = new ArrayList<>();
+    /** Reads multiple FASTQs one after the other. This is useful for when we have more than one FASTQ for a set of reads. */
+    class SequentialFastqReader implements Iterator<FastqRecord> {
+        private final List<FastqReader> readers = new ArrayList<>();
+        private int readerIndex = 0;
+
+        public SequentialFastqReader(final List<File> inputs) {
+            for (final File input : inputs) {
+                readers.add(new FastqReader(input, ALLOW_AND_IGNORE_EMPTY_LINES));
+            }
+        }
+
+        public boolean hasNext() {
+            while (readerIndex < readers.size()) {
+                if (readers.get(readerIndex).hasNext()) return true;
+                readerIndex++;
+            }
+            return false;
+        }
+
+        public FastqRecord next() {
+            if (readers.size() <= readerIndex) {
+                throw new NoSuchElementException();
+            }
+            return readers.get(readerIndex).next();
+        }
+
+        public void close() {
+            readers.stream().forEach(FastqReader::close);
+        }
+    }
+
+    /** Reads multiple FASTQs at the same time, ensuring their read names match along the way.  This is
+     * useful for when we don't have a single interleaved FASTQ. */
+    class ParallelFastqReader implements Iterator<List<FastqRecord>> {
+        public final List<SequentialFastqReader> readers = new ArrayList<>();
 
-        public MultiFastqReader(final File fastq, final File... fastqs) {
-            readers.add(new FastqReader(fastq, ALLOW_AND_IGNORE_EMPTY_LINES));
-            for (final File otherFastqs : fastqs) {
-                readers.add(new FastqReader(otherFastqs, ALLOW_AND_IGNORE_EMPTY_LINES));
+        @SafeVarargs
+        public ParallelFastqReader(final List<File> fastq, final List<File>... fastqs) {
```

Comment:
I feel like either here, or above, there should probably be some validation that you have the same number of fastqs in each list.